### DISCUSSION TITLE
logging chat ?

### DISCUSSION THREAD

**Discussion by strk** [jamulussoftware/jamulus#2988] [Ideas]
I'm experimenting with running my own server and wondering if there's a way to log the text going to the chat. It would be useful because there's a lot of links sent to the chat

**Comment by rdica** [comment]
You might be able to use the RPC interface of a connected client to collect chat text, afaik there isn't a means to do so from the server itself.

**Comment by pljones** [comment]
You would need to take legal responsibility for logging communications potentially considered private between users.  Jamulus itself does not monitor or log these messages.

**Reply by mcfnord** [reply]
A disclaimer should suffice.

**Comment by mcfnord** [comment]
@Rob-NY might have clues.

**Comment by Rob-NY** [comment]
I have already built (and shared demos) of RPC extensions that can do all sorts of things like this (see Experimental VULTR on Any Genre 1).   However, for them to become a part of core jamulus, the privacy policy needs to shift EVERYTHING to the server operators — which, frankly, keeps getting pulled back because this dev team wants to make privacy and security representations which it can’t (and shouldn’t) do.   @mcfnord is correct that a ‘good actor’ operator would state this in their welcome message - but even stating such would cause confusion with the direction of the core privacy policy as it stands now (live and in the new PR).   There has been tremendous push back on the introduction of any capabilities for RPC, Logging, Access Control, and even language that empower server operators to bolt on additional capabilities.

So, @strk, I would not expect to see this capability in core Jamulus anytime soon.  But if you can compile the server yourself, you have many more options.

And @pljones, your comment above is exactly why I think it is very dangerous ground to assume that every server out there is a clean compile from the repo.  Saying “… Jamulus itself does not…” is quite misleading as there is no way to really tell what the server you connect to is doing.   And that is my entire point in the other PR thread.

**Reply by pljones** [reply]
The term "Jamulus" is used to represent the unmodified software as downloaded.  Any modified software _is not Jamulus_ and should not be representing itself as such.

Life's just easier like this.

**Reply by mcfnord** [reply]
> and even language that empower server operators to bolt on additional capabilities.

What sort of language?

**Reply by Rob-NY** [reply]
> > and even language that empower server operators to bolt on additional capabilities.
> 
> What sort of language?

Language in the privacy policy that acknowledges that servers are modifiable, etc, etc...

**Reply by Rob-NY** [reply]
> The term "Jamulus" is used to represent the unmodified software as downloaded. Any modified software _is not Jamulus_ and should not be representing itself as such.
> 
> Life's just easier like this.

Unfortunately, this isn't how open source works. While I completely understand your conundrum of wanting to bundle up the software along with the ecosystem it runs within into an all-encompassing privacy policy, the reality is that you really will not be successful at doing that. In my opinion you should distance yourself from it -- and to a point that was done with the latest PP release -- but it didn't go far enough for my taste. But I respect your position even if I do not completely agree with it.
